File: server/src/models/Room.ts
import { Model, DataTypes } from 'sequelize';
import sequelize from '../db/connection';

class Room extends Model {
  declare id: number;
  declare name: string;
  declare passwordHash: string | null;
  declare createdTime: Date;
  declare lastActiveTime: Date;
}

Room.init({
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true
  },
  name: {
    type: DataTypes.STRING(100),
    allowNull: false
  },
  passwordHash: {
    type: DataTypes.STRING(255),
    allowNull: true
  },
  createdTime: {
    type: DataTypes.DATE(3),
    allowNull: false,
    defaultValue: DataTypes.NOW
  },
  lastActiveTime: {
    type: DataTypes.DATE(3),
    allowNull: false,
    defaultValue: DataTypes.NOW
  }
}, {
  sequelize,
  modelName: 'Room',
  tableName: 'rooms',
  timestamps: false
});

export default Room; 

File: server/internal/models/room.go
package models

import (
	"time"

	"gorm.io/gorm"
)

// Room represents a sync room
type Room struct {
	ID             uint           `gorm:"primaryKey;autoIncrement" json:"id"`
	Name           string         `gorm:"type:varchar(100);not null" json:"name"`
	PasswordHash   *string        `gorm:"type:varchar(255)" json:"passwordHash,omitempty"`
	CreatedTime    time.Time      `gorm:"not null;autoCreateTime:milli" json:"createdTime"`
	LastActiveTime time.Time      `gorm:"not null;autoUpdateTime:milli" json:"lastActiveTime"`
	DeletedAt      gorm.DeletedAt `gorm:"index" json:"-"`
}

// TableName specifies the table name for Room model
func (Room) TableName() string {
	return "rooms"
}
